internal/ethapi/simulate.go

```diff
@@ -81,6 +81,21 @@ type simBlockResult struct {
 	Calls       []simCallResult
 }
 
+type nullReceiptGetter struct{}
+
+func (nullReceiptGetter) GetReceipts(context.Context, common.Hash) (types.Receipts, error) {
+	return nil, nil
+}
+
+func (r *simBlockResult) MarshalJSON() ([]byte, error) {
+	blockData, err := RPCMarshalBlock(context.Background(), r.Block, true, r.fullTx, r.chainConfig, nullReceiptGetter{})
```

Comment:
Our RPCMarshalBlock implementation requires a context and backend. So I replaced the backend by a narrow interface, stubbed it out and threw the stub in to the call here.